hal/api/RawSerial.h

```diff
@@ -32,6 +32,8 @@ namespace mbed {
  * Can be used for Full Duplex communication, or Simplex by specifying
  * one pin as NC (Not Connected)
  *
+ * @Note Synchronization level: Not protected
```

Comment:
The lock and unlock are just virtual stubs.  This allows child classes to implement the lock as they would like.